python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -1239,8 +1238,248 @@ def asset_check_spec(self) -> AssetCheckSpec:
         return asset_checks_def.spec
 
 
-# actually forking the object type for assets is tricky for users in the cases of:
-#  * manually constructing ops to make AssetsDefinitions
-#  * having ops in a graph that form a graph backed asset
-# so we have a single type that users can call by their preferred name where appropriate
-AssetExecutionContext: TypeAlias = OpExecutionContext
+############################
+##### AssetExecutionContext
+############################
+
+# To preserve backwards compatibility, all of the methods on OpExecutionContext need to be accessible
+# from AssetExecutionContext. Over time, we would like to deprecate some of these methods and have users
+# access them through the op_execution_context stored in the AssetExecutionContext
+# The following sets/maps are used to determine which methods need deprecation warnings, and how to
+# direct users to the correct method to use
+
+OP_EXECUTION_CONTEXT_ONLY_METHODS = set(
+    [
+        "describe_op",
+        "file_manager",
+        "has_assets_def",
+        "get_mapping_key",
+        # "get_step_execution_context", # used by internals
```

Comment:
We should add these internal methods to a different set and emit a different warning